### ISSUE TITLE
Sometimes braces are removed too eagerly

### ISSUE THREAD

**Issue by pulkomandy** [owenca/haiku-format#13]
```
void
BView::_PrintTree()
{
	int32 spaces = 2;
	BView* c = fFirstChild; //c = short for: current
	printf( "'%s'\n", Name() );
	if (c != NULL) {
		while(true) {
			// action block
			{
				for (int i = 0; i < spaces; i++)
					printf(" ");

				printf( "'%s'\n", c->Name() );
			}

			// go deep
			if (c->fFirstChild) {
				c = c->fFirstChild;
				spaces += 2;
			} else {
				// go right
				if (c->fNextSibling) {
					c = c->fNextSibling;
				} else {
					// go up
					while (!c->fParent->fNextSibling && c->fParent != this) {
						c = c->fParent;
						spaces -= 2;
					}

					// that enough! We've reached this view.
					if (c->fParent == this)
						break;

					c = c->fParent->fNextSibling;
					spaces -= 2;
				}
			}
		}
	}
}
```

Gets reformatted to:

```
void
BView::_PrintTree()
{
	int32 spaces = 2;
	BView* c = fFirstChild; // c = short for: current
	printf("'%s'\n", Name());
	if (c != NULL)
		while (true) // action block
		{
			for (int i = 0; i < spaces; i++)
				printf(" ");

			printf("'%s'\n", c->Name());
		}

	// go deep
	if (c->fFirstChild) {
		c = c->fFirstChild;
		spaces += 2;
	} else // go right
		if (c->fNextSibling)
		c = c->fNextSibling;
	else {
		// go up
		while (!c->fParent->fNextSibling && c->fParent != this) {
			c = c->fParent;
			spaces -= 2;
		}

		// that enough! We've reached this view.
		if (c->fParent == this)
			break;

		c = c->fParent->fNextSibling;
		spaces -= 2;
	}
}
```

The whole "go deep" block of code is moved outside of the while and if.

Note: I'm using korli patch from #12, ported to clang11, and removed the latest commit which was even more broken.

**Comment by owenca** [comment]
Will disable it.

**Comment by owenca** [comment]
This issue is related to #5.

**Comment by owenca** [comment]
See https://github.com/owenca/haiku-format/issues/5#issuecomment-741035179.

**Comment by owenca** [comment]
Fixed in 88f7234.
